src/algorithm/butterflies.rs

```diff
@@ -3,20 +3,54 @@ use num_complex::Complex;
 use crate::{common::FftNum, FftDirection};
 
 use crate::array_utils;
-use crate::array_utils::{RawSlice, RawSliceMut};
 use crate::common::{fft_error_inplace, fft_error_outofplace};
 use crate::twiddles;
 use crate::{Direction, Fft, Length};
 
+pub(crate) trait LoadStore<T> {
+    unsafe fn load(&self, idx: usize) -> T;
+    unsafe fn store(&mut self, val: T, idx: usize);
+}
+
+impl<T: Copy> LoadStore<T> for [T] {
+    #[inline(always)]
+    unsafe fn load(&self, idx: usize) -> T {
+        *self.get_unchecked(idx)
```

Comment:
One thing the RawSlice struct had that these trait imps don't, is a debug_assert!() that the index is in range. I think they play an important part of maintaining peace of mind that unit tests will catch any out-of-bounds errors before they make it into a release. Could you re-add them?

Reply:
Certainly! I had incorrectly assumed that get_unchecked and get_unchecked_mut already had debug_asserts. Apparently I remembered the proposal for doing so without remembering that it wasn't approved!